tests/test_root_model.py

```diff
@@ -325,17 +328,54 @@ class Model(RootModel[int]):
             model_config = ConfigDict(extra=extra_value)
 
 
-@pytest.mark.xfail(reason='TODO: support applying defaults with `RootModel`')
 def test_root_model_default_value():
     class Model(RootModel):
         root: int = 42
 
     m = Model()
     assert m.root == 42
-    assert m.model_dump == 42
+    assert m.model_dump() == 42
+    assert m.__pydantic_fields_set__ == set()
+
+
+def test_root_model_default_factory():
+    class Model(RootModel):
+        root: int = Field(default_factory=lambda: 42)
+
+    m = Model()
+    assert m.root == 42
+    assert m.model_dump() == 42
+    assert m.__pydantic_fields_set__ == set()
 
 
-def test_root_model_as_attr_with_validate_defaults():
+def test_root_model_wrong_default_value_without_validate_default():
+    class Model(RootModel):
+        root: int = '42'
+
+    assert Model().root == '42'
+
+
+def test_root_model_default_value_with_validate_default():
+    class Model(RootModel):
+        model_config = ConfigDict(validate_default=True)
+
+        root: int = '42'
+
+    m = Model()
+    assert m.root == 42
+    assert m.model_dump() == 42
```

Comment:
Can you add an assertion for the `__pydantic_fields_set__` here and below?

Reply:
Added and found a bug:)